schema-engine/connectors/sql-schema-connector/src/sql_doc_parser.rs

```diff
@@ -237,17 +285,18 @@ fn validate_param(param: &ParsedParameterDoc<'_>, input: Input<'_>) -> Connector
     Ok(())
 }
 
-fn parse_param(param_input: Input<'_>) -> ConnectorResult<ParsedParameterDoc<'_>> {
+fn parse_param<'a>(param_input: Input<'a>, enum_names: &'a [String]) -> ConnectorResult<ParsedParameterDoc<'a>> {
     let input = param_input.strip_prefix_str("@param").unwrap().trim_start();
 
-    let (input, typ) = parse_typ_opt(input)?;
+    let (input, typ) = parse_typ_opt(input, enum_names)?;
     let (input, position) = parse_position_opt(input)?;
-    let (input, alias) = parse_alias_opt(input)?;
+    let (input, alias, nullable) = parse_alias_opt(input)?;
     let documentation = parse_rest(input)?;
 
     let mut param = ParsedParameterDoc::default();
 
     param.set_typ(typ);
+    param.set_nullable(nullable.then_some(true));
```

Comment:
make this clearer

Reply:
I ended up keeping it as an `Option`. The rationale is that I'd like to keep the parser unopinionated. If it finds a `?`, it's `Some(true)`. Else, it's `None`. It's up to the `introspect_sql` implementation to give semantic value to this information, and `.unwrap_or(default)` with the correct default. I did simplify things by always returning `None` when `?` is not present. Previously, we returned `Some(false)` when `?` was not present.